### PR TITLE
Add support for new lock syntax in MySQL8

### PR DESCRIPTION
**Pull request by patrickcarnahan**
## Description

This completes the work to enable the following syntax in vitess:

* `SELECT .. FOR SHARE (NOWAIT|SKIP LOCKED)`
* `SELECT .. FOR UPDATE (NOWAIT|SKIP LOCKED)`

The aim is to finish the work started in https://github.com/vitessio/vitess/pull/5336, which added the keywords to the syntax but didn't modify the ast and other functions to handle it.

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/5249

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [X] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/sqlparser/testdata/union_cases.txt [hunk 1]
```diff
@@ -1004,7 +1004,7 @@ INPUT
 SELECT 1 FOR SHARE UNION SELECT 2;
 END
 ERROR
-syntax error at position 19 near 'SHARE'
```
Comment (patrickcarnahan) [inline go/vt/sqlparser/testdata/union_cases.txt:1007]: this actually matches the error mysql8 provides when issuing the same query via explain:

```
mysql> explain select 1 for share union select 2;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'union select 2' at line 1
```

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -6868,6 +6868,26 @@ FOR UPDATE
   {
     $$ = ForUpdateLock
   }
+| FOR UPDATE NOWAIT
```
Comment (patrickcarnahan) [inline go/vt/sqlparser/sql.y:6871]: not sure if there is a "better" way to do this, but i saw this type of pattern elsewhere so seemed to make sense.

**Review by patrickcarnahan — COMMENTED** [review]
i also found `query_tests.go` under `go/vt/vttablet/endtoend/queries_test.go` which i can add to but wasn't sure if that is relevant for this type of change.

**Comment by arthurschreiber** [comment]
I'm going to merge this as we tested this out and it was working as expected. 👍

**Pull request merged**